### DISCUSSION TITLE
[Question] Auto Highlight selection item when Click item

### DISCUSSION THREAD

**Discussion by QiubyZ** [hrsh7th/nvim-cmp#2054] [General]
Hello, I am a new user in vim, I am also new to this thing called cmp-vim.

On my android, I use <ScrollwheelUp> and <ScrollWheelDown> to navigate the autocomplete options on cmp.
as follows.

https://github.com/user-attachments/assets/595c5826-678b-4a54-b76c-b2b20d9af06d

my question:
Is it possible or does anyone know how to make this click work directly without having to highlight it first?
I am using the code implementation below, it seems that it never selects the item in the autocomplete "unless" it is highlighted.

completeopt=menu,menuone,noselect
mouse=a
```lua
        ['<LeftMouse>'] = cmp.mapping(function(fallback)
      if cmp.visible() then
            local entry = cmp.get_selected_entry()
            if entry then
              vim.notify(entry:get_completion_item().label)
              fallback()
              
            else

              vim.notify("No selected")
              fallback()
            end
      else
          fallback()
      end
        end, { 'i', 's' }),
```

https://github.com/user-attachments/assets/5a565cbc-cf32-4f30-8415-8fb4a2a9f0a4

This question came up because I saw Neotree which can be selected, like this

https://github.com/user-attachments/assets/ec745113-e595-477c-b550-9ac33fad08c3

maybe someone can help me to implement it into cmp

**Comment by Shougo** [comment]
Unfortunately, you cannot.
Because the completion window is not the normal window.
nvim-cmp cannot know what the item is selected when clicked.
nvim-tree works because it is normal buffer.

**Reply by QiubyZ** [reply]
> Unfortunately, you cannot. Because the completion window is not the normal window. nvim-cmp cannot know what the item is selected when clicked. nvim-tree works because it is normal buffer.

Is there an update the next day or is this just here to stay?

**Reply by Shougo** [reply]
~~It cannot implement the feature.~~

First and foremost, the use of a mouse is not recommended.
Vim is not VSCode.

**Reply by Shougo** [reply]
You can get current mouse position by `getmousepos()`.
But it is not converted to selected item.

**Reply by QiubyZ** [reply]
> ~It cannot implement the feature.~
> 
> First and foremost, the use of a mouse is not recommended. Vim is not VSCode.

if your statement is like that, it means vim absolutely must use the keyboard.
but that would raise further questions, like if that's the case, why did they make a file manager or UI that can be clicked using a mouse?
What I mean is that with the development of current technology, the mouse and touch screen or Android that now has Termux has similarities to the terminal on Windows/Linux OS but why is it still limited with words "mouse not recommended"
this makes me confused

**Reply by Shougo** [reply]
> but that would raise further questions, like if that's the case, why did they make a file manager or UI that can be clicked using a mouse?

Hm....

It is due to historical reasons. This is a long story, but Vim was not created as a text editor based on a GUI. Vim was designed to work in a CUI. Although the current Vim can use the mouse to some extent, it is primarily based on keyboard operations, which can be inconvenient if you want to use mouse.

**Reply by QiubyZ** [reply]
> > but that would raise further questions, like if that's the case, why did they make a file manager or UI that can be clicked using a mouse?
> 
> Hm....
> 
> It is due to historical reasons. This is a long story, but Vim was not created as a text editor based on a GUI. Vim was designed to work in a CUI. Although the current Vim can use the mouse to some extent, it is primarily based on keyboard operations, which can be inconvenient if you want to use mouse.

This is not an excuse for being lazy to update it, right? 🤣

**Reply by Shougo** [reply]
Current nvim-cmp doesn't support the mouse.
You can request, but the author seems busy and don't want to implement new feature now.
